### PR TITLE
How to guide overview

### LINKED ISSUE #896: Update and enhance how-to-guide section
This issue is to track :
* What has been done and 
* What we are on currently working :)

## Team members
Abhishek Jaiswal - @abhishek-iiit 
Vasvi Sood  - @vasvi-sood 

# References 
- [x] **Readme(to github section)** 
- [x] **Overview** 
- [x] **Add events to the bot**
    - [x] Add events and customise a bot
- [x] **Add more functionality to your bot**
    - [x] Dealing with messages
    - [x] Manage Contacts
    - [x] Manage Rooms
    - [x] Send and Accept Friend Requests
    - [x] Sending and Receiving files
- [x] **Deploy on IM platfrom**
    - [x] Deploy in Wechat
    - [x] Deploy in Whatsapp
    - [x] Deploy in Lark
    - [x] Deploy in Gitter
    - [x] Deploy in Wechat Official Account
    - [x] Deploy in PadLocal
    - [x] Deploy in WeChat4U
- [x] **Overview with containers**
    - [x] Deploy with Heroku
    - [x] Deploy with Docker

Please link pull request to this section for the easy access.

### PR DESCRIPTION
**Pull request by abhishek-iiit**
## Added Overview Section
Screenshot from 2021-08-24 12-27-54
Screenshot from 2021-08-24 12-28-02

### CODE REVIEW

**Review by Rohitesh-Kumar-Jain — APPROVED** [review]
Perfect PR, pls link to the PR that will be closed if we merge this.

**Review by proudofsimin — APPROVED** [review]
approved as already reviewed in #901.

**Comment by lijiarui** [comment]
Hi @abhishek-iiit , As GSoD Project PR Review Rules in https://github.com/wechaty/wechaty.js.org/issues/1045

This PR should follow our rules and get enough approval, then I will review this PR. So I remove the `ready to merge` tag and feel free to add this tag when it gets enough approval.

**Comment by Rohitesh-Kumar-Jain** [comment]
@vasvi-sood  We would be glad if you may once go through the PR, and either request for changes or approve the PR.

**Review by vasvi-sood — APPROVED** [review]
This PR looks Great!

File: docusaurus/docs/howto/overview.mdx [hunk 1]
```diff
@@ -6,33 +6,60 @@ sidebar_label: 'How-to Guides: Overview'
```
Comment (lijiarui) [inline docusaurus/docs/howto/overview.mdx:6]: I think we should change the title from ` Wechaty How-to guides overview` to `Overview`, change sidebar_label from `How-to Guides: Overview` to `Overview`

**Review by lijiarui — CHANGES_REQUESTED** [review]

File: docusaurus/docs/howto/overview.mdx (hunk 1 above)
Comment (abhishek-iiit) [reply docusaurus/docs/howto/overview.mdx:6]: Thankyou, changes made as suggested :)

File: docusaurus/docs/howto/overview.mdx [hunk 2]
```diff
@@ -1,38 +1,65 @@
...
+sidebar_label: 'Overview'
 ---
 
 <!-- how-to/cookbook - Explains how to accomplish a specific task. -->
 
-How-to guides are recipes. They guide you through the steps involved in addressing key problems and use-cases. They are more advanced than tutorials and assume some knowledge of how Wechaty works.
+How-to guides are like recipes when you are cooking. Equipped with basic knowledge about Wechaty, you can find key problems and use-cases at [Wechaty Examples](https://wechaty.js.org/docs/examples/overview). If you are a total beginner, you can go to [Wechaty getting started](https://wechaty.js.org/docs/getting-started/) first.
 
-These are some use cases and code snippets to get you started with Redux in a real app. They assume you understand the topics in [basic](#b) and [advanced](#a) tutorials.
+<!-- These are some use cases and code snippets to get you started with Redux in a real app. They assume you understand the topics in [basic](#b) and [advanced](#a) tutorials. They are more advanced than tutorials and assume some knowledge of how Wechaty works. -->
```
Comment (lijiarui) [inline docusaurus/docs/howto/overview.mdx:11]: It seems you comment on this part and it wouldn't be shown in the docs.

File: docusaurus/docs/howto/overview.mdx [hunk 3]
```diff
@@ -1,38 +1,65 @@
...
+- [Deploy in WeChat4U](#)
+
+## Deploy with containers
+
+Deploy the bot with highly popular container solutions
+
+- [Deploy with Heroku](#)
+- [Deploy with Docker](#)
+
+<!-- ## Using Puppets with Wechaty (How to switch between puppets) -->
```
Comment (lijiarui) [inline docusaurus/docs/howto/overview.mdx]: It seems you make them as docs comment here, and it cannot be shown in the docs.

I think this part should be shown in the docs.

**Review by lijiarui — DISMISSED** [review]

File: docusaurus/docs/howto/overview.mdx (hunk 2 above)
Comment (abhishek-iiit) [reply docusaurus/docs/howto/overview.mdx:11]: Yes, the line 11 and the line 9 explains the same process. So, made to comment out the later part :)

**Review by huan — APPROVED** [review]
LGTM.

Please read and follow #1257, and feel free to help yourself to merge it by following the merge workflow.

Thank you very much!

**Comment by Rohitesh-Kumar-Jain** [comment]
@abhishek-iiit pls remove the merge conflicts

**Pull request merged**